docs/content/migration.md

```diff
@@ -0,0 +1,545 @@
+with theming. The use of Sass also moves the theming system to a more well-supported and
+broadly-understood toolchain.
+
+#### Defining palettes
+
+AngularJS Material supports defining [custom palettes][151] or [extending existing palettes][158].
+
+In Angular Material, custom palette creation, including definition of hues and contrast colors, can
+be accomplished by copying one of the [provided palettes][167], modifying it, and then using it in
+your [custom theme creation][164].
+
+#### Defining themes
+
+AngularJS Material uses a provider, [`$mdThemingProvider`][144], to define the theme configuration
+used to generate styles during application loading. This provider also includes a default theme for
+cases when no custom palettes are provided.
+
+Angular Material has no default theme; a theme CSS file must _always_ be included. You can either
+use one of the [pre-built theme files][163], or define a [custom theme][164] with Sass mixins,
+including the output in your application just like any other CSS asset.
+
+##### Defining themes at run-time
+
+AngularJS Material uses a service, [`$mdTheming`][143], to define themes and generate CSS styles at
+run-time.
+
+Defining and generating theme CSS at run-time is not supported by Angular Material. However,
+loading additional theme CSS into your application (or modifying an existing theme's `link` node
+to [swap theme CSS files][181]) at run-time is supported.
+
+Your application can choose to generate additional themes at compile-time or build an API to
+dynamically generate theme CSS on-demand. In either case, it is possible to [lazy load][165] the
+theme CSS at run-time.
+
+#### Applying themes to your application
+
+AngularJS Material has a custom directive, [`md-theme`][155], to apply a theme to a specific
+component and its children.
+
+Rather than providing a special directive API, Angular Material's Sass-based theming system relies
+on standard Sass mixins and vanilla CSS selectors. For example, you can customize an individual
+button (or set of buttons) by defining and applying a class:
+
+```scss
+// Define a CSS class for a special button treatment.
+.super-special-button {
+  @include mat-button-theme($super-special-theme);
+}
+```
+```html
+<!-- Apply this treatment the same as you would with any other CSS class. -->
+<button mat-button class="super-special-button">Save</button>
+```
+
+Just like any other CSS, these classes can be applied to any element in your application and can
+be added/removed/toggled to swap colors at run-time.
+
+[See the full documentation](https://material.angular.io/guide/theming) for details on themes,
+including examples of multiple themes.
+
+##### Dynamically applying themes
+
+AngularJS Material's [`md-theme`][155] and [`md-theme-watch`][156] directives support data binding
+like `md-theme="{{ dynamicTheme }}" md-theme-watch`.
+
+With Angular Material, you can make use of Angular's [ngClass][159] API to dynamically apply classes
```

Comment:
I'd emphasize a normal class binding over ngClass rather than the other way around

Reply:
I agree in general and considered that, but the `ngClass` example is the closest to the functionality of `md-theme="{{ dynamicTheme }}" md-theme-watch` unless there is a `[class.xxx]="yyy"` syntax that I'm forgetting which allows changing between multiple classes.